OpenRA.Mods.Common/Widgets/Logic/Lobby/LobbyOptionsLogic.cs

```diff
@@ -0,0 +1,179 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2017 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using OpenRA.Network;
+using OpenRA.Traits;
+using OpenRA.Widgets;
+
+namespace OpenRA.Mods.Common.Widgets.Logic
+{
+	public class LobbyOptionsLogic : ChromeLogic
+	{
+		readonly ScrollPanelWidget panel;
+		readonly Widget optionsContainer;
+		readonly Widget checkboxTemplate;
+		readonly Widget dropdownTemplate;
+
+		readonly Func<MapPreview> getMap;
+		readonly OrderManager orderManager;
+		readonly Func<bool> configurationDisabled;
+		MapPreview mapPreview;
+		bool validOptions;
+
+		[ObjectCreator.UseCtor]
+		internal LobbyOptionsLogic(Widget widget, OrderManager orderManager, Func<MapPreview> getMap, Func<bool> configurationDisabled)
+		{
+			this.getMap = getMap;
+			this.orderManager = orderManager;
+			this.configurationDisabled = configurationDisabled;
+
+			panel = (ScrollPanelWidget)widget;
+			optionsContainer = widget.Get("LOBBY_OPTIONS");
+			optionsContainer.IsVisible = () => validOptions;
+			checkboxTemplate = optionsContainer.Get("CHECKBOX_ROW_TEMPLATE");
+			dropdownTemplate = optionsContainer.Get("DROPDOWN_ROW_TEMPLATE");
```

Comment:
Please rename these to `*RowTemplate` for easier readability (I may be sleepy, but it took more than it should have to understand the below code).